js/flexbuffers.js

```diff
@@ -0,0 +1,1015 @@
+      return finished;
+    },
+    addKey: function(key) {
+      integrityCheckOnKeyAddition();
+      writeKey(key);
+    },
+    addInt: function(value, indirect = false, cache = false) {
+      integrityCheckOnValueAddition();
+      if (!indirect) {
+        stack.push(intStackValue(value));
+        return;
+      }
+      if (cache && indirectIntCache.hasOwnProperty(value)) {
+        stack.push(indirectIntCache[value]);
+        return;
+      }
+      const stackValue = intStackValue(value);
+      const byteWidth = align(stackValue.width);
+      const newOffset = computeOffset(byteWidth);
+      const valueOffset = offset;
+      stackValue.writeToBuffer(byteWidth);
+      const stackOffset = offsetStackValue(valueOffset, flexbuffers.ValueType.INDIRECT_INT, stackValue.width);
+      stack.push(stackOffset);
+      offset = newOffset;
+      if (cache) {
+        indirectIntCache[value] = stackOffset;
+      }
+    },
+    addUInt: function(value, indirect = false, cache = false) {
+      integrityCheckOnValueAddition();
+      if (!indirect) {
+        stack.push(uintStackValue(value));
+        return;
+      }
+      if (cache && indirectUIntCache.hasOwnProperty(value)) {
+        stack.push(indirectUIntCache[value]);
+        return;
+      }
+      const stackValue = uintStackValue(value);
+      const byteWidth = align(stackValue.width);
+      const newOffset = computeOffset(byteWidth);
+      const valueOffset = offset;
+      stackValue.writeToBuffer(byteWidth);
+      const stackOffset = offsetStackValue(valueOffset, flexbuffers.ValueType.INDIRECT_UINT, stackValue.width);
+      stack.push(stackOffset);
+      offset = newOffset;
+      if (cache) {
+        indirectUIntCache[value] = stackOffset;
+      }
+    },
+    addFloat: function(value, indirect = false, cache = false) {
+      integrityCheckOnValueAddition();
+      if (!indirect) {
+        stack.push(floatStackValue(value));
+        return;
+      }
+      if (cache && indirectFloatCache.hasOwnProperty(value)) {
+        stack.push(indirectFloatCache[value]);
+        return;
+      }
+      const stackValue = floatStackValue(value);
+      const byteWidth = align(stackValue.width);
+      const newOffset = computeOffset(byteWidth);
+      const valueOffset = offset;
+      stackValue.writeToBuffer(byteWidth);
+      const stackOffset = offsetStackValue(valueOffset, flexbuffers.ValueType.INDIRECT_FLOAT, stackValue.width);
+      stack.push(stackOffset);
+      offset = newOffset;
+      if (cache) {
+        indirectFloatCache[value] = stackOffset;
+      }
+    },
+    startVector: function() {
+      startVector();
+    },
+    startMap: function() {
+      startMap();
+    },
+    end: function() {
+      end();
+    }
+  };
+};
+
+flexbuffers.encode = (object, size = 2048) => {
+  const builder = flexbuffers.builder(size > 0 ? size : 2048);
+  builder.add(object);
+  return builder.finish();
+};
+
+function fromUTF8Array(data) { // array of bytes
```

Comment:
JS does not have a function for this?

Reply:
Core JS doesn't, but both Web and Node.js have own built-in methods. Best to use them instead, with feature-detection when library is loaded.

Reply:
Actually as far as I understand TextDecoder / Encoder is now also part of node.js so I am using it instead now.